src/throttle.rs

```diff
@@ -0,0 +1,65 @@
+use tokio::sync::mpsc::Receiver;
+use tokio::time;
+
+use crate::GooseConfiguration;
+
+/// This throttle thread limits the maximum number of requests that can be made across
+/// all GooseUser threads. When enabled, GooseUser threads must add a token to the
+/// bounded channel before making a request, and this thread limits how frequently
+/// tokens are removed thereby throttling how fast requests can be made. It is a variation
+/// on the leaky bucket algorithm: instead of leaking the overflow we asynchronously block.
+pub async fn throttle_main(
+    configuration: GooseConfiguration,
```

Comment:
As the only thing used from the configuration struct is `throttle_requests` and the callee more clearly is ok to `unwrap()` its value , I'd say this parameter should be instead be an usize and be called as `throttle_main(configuration.throttle_requests.unwrap(), ....` in lib.rs

Reply:
Nice little cleanup. The parent has already unwrapped the value, and there's no need to clone the entire configuration struct -- fixed.